samples/04_gis_analysts_data_scientists/address-standardization-and-correction-with-SequenceToSequence.ipynb

```diff
@@ -0,0 +1 @@
oyth rhodeisland aveune, mason city, iowa, 50401, us',\n", "  '220, s rhode island ave, mason city, ia, 50401, us')]"]}, "execution_count": 22, "metadata": {}, "output_type": "execute_result"}], "source": ["model.predict(txt, num_beams=6, max_length=50)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Conclusion"]}, {"cell_type": "markdown", "metadata": {}, "source": ["In this notebook we will build an address standardization and correction model using `SequenceToSequence` class of `arcgis.learn.text` module. The dataset consisted of a pair of non-standard, incorrect (synthetic errors) house addresses and corresponding correct, standard house addresses from the United States. To achieve this we used a [t5-base](https://huggingface.co/t5-base) pretrained transformer to build a SequenceToSequence model to standardize and correct the input house addresses. Below are the results on sample inputs."]}, {"cell_type": "markdown", "metadata": {}, "source": ["<span style=\"background-color: LightSalmon\">Non-Standard</span> &rarr;  <span style=\"background-color: LightGreen\">Standard\n", "</span>  ,  <span style=\"background-color: LightCoral\">Error</span> &rarr; <span style=\"background-color: LightGreen\">Correction\n", "</span>\n", "* <span style=\"font-size:1.5em;\">940, <span style=\"background-color: LightSalmon;\">north</span> pennsylvania <span style=\"background-color: LightSalmon\">avneue</span>, mason <span style=\"background-color: LightCoral\">icty</span>, <span style=\"background-color: LightSalmon\">iowa</span>, 50401, us &rarr; 940, <span style=\"background-color: LightGreen\">n</span> pennsylvania <span style=\"background-color: LightGreen\">ave</span>, mason <span style=\"background-color: LightGreen\">city</span>, <span style=\"background-color: LightGreen\">ia</span>, 50401, us</span>\n", "\n", "\n", "* <span style=\"font-size:1.5em;\">220, <span style=\"background-color: LightCoral\">soyth rhodeisland</span> <span style=\"background-color: LightSalmon\">aveune</span>, mason city, <span style=\"background-color: LightSalmon\">iowa</span>, 50401, us &rarr; 220, <span style=\"background-color: LightGreen\">s rhode island ave</span>, mason city, <span style=\"background-color: LightGreen\">ia</span>, 50401, us</span>\n", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# References"]}, {"cell_type": "markdown", "metadata": {}, "source": ["[1] [Learning Rate](https://en.wikipedia.org/wiki/Learning_rate)\n", "\n", "[2] [Accuracy](https://en.wikipedia.org/wiki/Accuracy_and_precision)\n", "\n", "[3] [Bleu score](https://en.wikipedia.org/wiki/BLEU)"]}], "metadata": {"kernelspec": {"display_name": "deep-learning", "language": "python", "name": "deep-learning"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.7.9"}}, "nbformat": 4, "nbformat_minor": 4}
```

Comment:
<p>Replace data path with this code <strong>Path(os.path.join(os.path.splitext(filepath)[0]))</strong></p><p>Replace <strong>DATA_ROOT </strong>with <strong>data_root</strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/870/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
done